src/CommentPolicy.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Laravelista\Comments;
+
+use Laravelista\Comments\Comment;
+
+class CommentPolicy
+{
+    /**
+     * Can user create the comment
+     *
+     * @param $user
+     * @return bool
+     */
+    public function create($user) : bool
+    {
+        return true;
+    }
+
+    /**
+     * Can user delete the comment
+     *
+     * @param $user
+     * @param Comment $comment
+     * @return bool
+     */
+    public function delete($user, Comment $comment) : bool
+    {
+        return $user->id === $comment->commenter_id && $comment->children->isEmpty();
```

Comment:
Maybe remove `&& $comment->children->isEmpty();` condition. I like the ability for the user to delete his own comment (and all replies to his comment by doing so) no matter what.

Maybe the user wrote something stupid and then someone wrote to him that he is stupid. With this condition the user would be unable to delete his comment. He can only edit his comment, but then the comment bellow that he is stupid would still be there.

This was just an example, but you get what I mean.

Reply:
Yes I agree. Actually that is a specific requirements for application which I am currently working.

Reply:
Yes I agree. Actually that is a specific requirements for application which I am currently working.